Regenerate all ANTLR parsers using version 4.11.1

knutwannheden opened the pull request:
Fixes #1615

knutwannheden commented on rewrite-hcl/build.gradle.kts:
```diff
@@ -22,7 +22,7 @@ dependencies {
     compileOnly(project(":rewrite-test"))
 
-    implementation("org.antlr:antlr4:4.9.+")
+    implementation("org.antlr:antlr4:4.11.1")
```
When I changed to `4.11.+` it started using 4.11.2-SNAPSHOT. I don't know enough Gradle yet to figure this out, which is why I changed it to `4.11.1`.

knutwannheden reviewed (commented):
AFAICT the files in `rewrite-hcl/src/main/antlr/.antlr` don't serve any purpose, so I deleted these (generated with ANTLR 4.8).

knutwannheden commented:
@traceyyoshima You reported that ANTLR 4.10 doesn't support Java 8: https://github.com/openrewrite/rewrite/pull/1616#issuecomment-1095539029. That was true for a short time after the release, but that was accidental and has been fixed in the meantime. (Do we have automated tests for this or do I manually need to check this?)

sambsnyd commented:
@knutwannheden we run the java tck on jdk 8, 11, and 17. So I would expect failures out of rewrite-java-8 if this did not support jdk8

traceyyoshima commented:
> @traceyyoshima You reported that ANTLR 4.10 doesn't support Java 8: [#1616 (comment)](https://github.com/openrewrite/rewrite/pull/1616#issuecomment-1095539029). That was true for a short time after the release, but that was accidental and has been fixed in the meantime. (Do we have automated tests for this or do I manually need to check this?)

Yes, we fixed the java version issue [here](https://github.com/antlr/antlr4/pull/3648).

The main issue blocking the changes was [dependencies](https://github.com/openrewrite/rewrite/issues/1615#issuecomment-1101688655) that exist on the SAAS that were stuck on an older version of ANTLR, which needed to be updated too. The dependencies have since been updated, but the work was not prioritized. We prioritized keeping the SAAS stable while the team made many heavy changes.

@sambsnyd If I remember correctly, these updates must be coordinated with the SAAS team; otherwise, version conflicts may cause outages.

knutwannheden commented:
> Yes, we fixed the java version issue [here](https://github.com/antlr/antlr4/pull/3648).

Cool. I didn't know that the bug was discovered and fixed by you guys!
